includes/widget-css-classes.class.php

```diff
@@ -15,94 +15,287 @@
 class WCSSC {
 
 	/**
+	 * Default capability to display the WCC form in widgets.
+	 * @static
+	 * @since  1.5.0
+	 * @var    string
+	 */
+	private static $default_cap = 'edit_theme_options';
+
+	/**
 	 * Adds form fields to Widget
 	 * @static
-	 * @param $widget
-	 * @param $return
-	 * @param $instance
+	 * @param  WP_Widget $widget
+	 * @param  mixed     $return
+	 * @param  array     $instance
 	 * @return array
-	 * @since 1.0
+	 * @since  1.0
 	 */
 	public static function extend_widget_form( $widget, $return, $instance ) {
 		$instance = wp_parse_args( $instance, array(
-			'classes' => null,
+			'ids' => '',
+			'classes' => '',
 			'classes-defined' => array(),
 		) );
 
+		/**
+		 * Change the capability to access the CSS ID field.
+		 *
+		 * @since  1.5.0
+		 * @param  string
+		 * @return string
+		 */
+		$access_id = current_user_can( apply_filters( 'widget_css_classes_id_input_capability', self::$default_cap ) );
+
+		/**
+		 * Change the capability to access the CSS Classes field.
+		 *
+		 * @since  1.5.0
+		 * @param  string
+		 * @return string
+		 */
+		$access_class = current_user_can( apply_filters( 'widget_css_classes_class_input_capability', self::$default_cap ) );
+
+		/**
+		 * Change the capability to access the predefined CSS Classes select field.
+		 * NOTE: If the user cannot access the predefined classes the regular input field is disabled as well.
+		 *
+		 * @since  1.5.0
+		 * @param  string
+		 * @return string
+		 */
+		$access_predefined = current_user_can( apply_filters( 'widget_css_classes_class_select_capability', self::$default_cap, $access_class ) );
+		if ( ! $access_predefined ) {
+			$access_class = false;
+		}
+
 		$fields = '';
 
-		if ( 1 === (int) WCSSC_Loader::$settings['show_id'] || WCSSC_Loader::$settings['type'] > 0 ) {
+		if ( WCSSC_Lib::get_settings( 'show_id' ) || WCSSC_Lib::get_settings( 'type' ) > 0 ) {
 			//$fields .= "<div class='wcssc' style='border: 1px solid #ddd; padding: 5px; background: #fafafa; margin: 1em 0; line-height: 1.5;'>\n";
-			$fields .= "<div class='wcssc'>\n";
+			$fields .= "<div class='wcssc' style='clear: both; margin: 1em 0;'>\n";
 
 			// show id field
-			if ( 1 === (int) WCSSC_Loader::$settings['show_id'] ) {
-				if ( ! isset( $instance['ids'] ) ) {
-					$instance['ids'] = null;
+			if ( WCSSC_Lib::get_settings( 'show_id' ) ) {
+				if ( $access_id ) {
+					$field = self::do_id_field( $widget, $instance );
+				} else {
+					$field = self::do_hidden( $widget->get_field_name( 'ids' ), $instance['ids'] );
 				}
-				$fields .= "\t<p><label for='widget-{$widget->id_base}-{$widget->number}-ids'>" . apply_filters( 'widget_css_classes_id', esc_html__( 'CSS ID', 'widget-css-classes' ) ) . ":</label>
-				<input type='text' name='widget-{$widget->id_base}[{$widget->number}][ids]' id='widget-{$widget->id_base}-{$widget->number}-ids' value='{$instance['ids']}' class='widefat' /></p>\n";
+				$fields .= "\t" . $field . "\n";

```

Comment:
@cleverness 
I was wondering, why do you add `\t` and `\n` everywhere? Did anyone file some sort of bug report for this?
From what I know this is not needed at all. Let me know!